doc/guides/porting/board_porting.rst

```diff
@@ -45,135 +31,333 @@ available in the source tree:
-SoCs belonging to the same family.
+Once you've found an existing board that uses your SoC, you can usually start
+by copy/pasting its board directory and changing its contents for your
+hardware.
 
-SoC
----
+You need to give your board a unique name. Run ``west boards`` for a list of
+names that are already taken, and pick something new. Let's say your board is
+called ``plank`` (please don't actually use that name).
 
-Finally, an SoC is actual hardware component that is physically available on a
-board.
+Start by creating the board directory ``zephyr/boards/<ARCH>/plank``, where
+``<ARCH>`` is your SoC's architecture subdirectory. (You don't have to put your
+board directory in the zephyr repository, but it's the easiest way to get
+started. See :ref:`custom_board_definition` for documentation on moving your
+board directory to a separate repository once it's working.)
 
-Board
-=====
+Your board directory should look like this:
 
-A board implements an SoC with all its features, together with peripherals
-available on the board that differentiates the board with additional interfaces
-and features not available in the SoC.
+.. code-block:: none
 
-A board port on Zephyr typically consists of two parts:
+   boards/<ARCH>/plank
+   ├── board.cmake
+   ├── doc
+   │   └── index.rst
+   ├── Kconfig.board
+   ├── Kconfig.defconfig
+   ├── plank_defconfig
+   ├── plank.dts
+   └── plank.yaml
 
-- A hardware description (usually done by device tree), which specifies embedded
-  SoC reference, connectors and any other hardware components such as LEDs,
-  buttons, sensors or communication peripherals (USB, BLE controller, ...).
+Replace ``plank`` with your board's name, of course.
 
-- A software configuration (done using Kconfig) of features and peripheral
-  drivers. This default board configuration is subordinated to features
-  activation which is application responsibility. Though, it should also enable
-  a minimal set of features common to many applications and to applicable
-  project-provided :ref:`samples-and-demos`.
+The mandatory files are:
 
+#. :file:`plank.dts`: a hardware description in :ref:`devicetree` format. This
+   declares your SoC, connectors, and any other hardware components such as
+   LEDs, buttons, sensors, or communication peripherals (USB, BLE controller,
+   etc).
+
+#. :file:`Kconfig.board`, :file:`Kconfig.defconfig`, :file:`plank_defconfig`:
+   software configuration in :ref:`kconfig` formats. This provides default
+   settings for software features and peripheral drivers.
+
+The optional files are:
+
+- :file:`board.cmake`: used for :ref:`flash-and-debug-support`. You can also
+  use this to build source code files whenever your board is used; see
+  :ref:`build_overview` for details.
+- :file:`index.rst`: documentation for your board. You only need this if you're
+  going to contribute the board to Zephyr.
+- :file:`plank.yaml`: a YAML file with miscellaneous metadata used by the
+  :ref:`sanitycheck_script`.
 
```

Comment:
I notice you do not include pinmux.c here. I think a mention and a link to the appropriate documentation would be appropriate, since it is included in almost every ARM board.

Reply:
Well, it's not true on the boards my company makes, because we do our pinmux in DT the way the good Lord intended ;).

Your point is well taken, thank you. I had deliberately left this out because we're trying to get away from pinmux.c in general and replace everything with devicetree, but that doesn't work yet and that confuses people, so we might as well document the current state of the world.